### ISSUE TITLE
Include a developer mode serve command

### ISSUE THREAD

**Issue by ang-zeyu** [MarkBind/markbind#1008]
**Is your request related to a problem?**

It may be helpful to have some developer only code, e.g.
- performance logging of various stages of site generation
- print extra debug statements

**Describe the solution you'd like**
A separate command, such as `markbind servedev` can be implemented.

It would accept some options like `-t` for more detailed performance logging, `-d` for printing debugging statements, etc.

Cons to note:
- code pollution? but it may serve as self-documenting code as well.

**Additional context**
Came up with this when developing the front end theme switcher to quickly test changes through all themes ( https://github.com/MarkBind/markbind/pull/999 ).

I can only think of a few use cases now, but I'd like to think the potential uses for this are many.

**Comment by yamgent** [comment]
What kind of debugging statements were you intending to print? The logging feature should be enhanced first by allowing the level of logging to be specified in command line (e.g. `markbind serve --log=DEBUG` or something).

**Comment by ang-zeyu** [comment]
> What kind of debugging statements were you intending to print? The logging feature should be enhanced first by allowing the level of logging to be specified in command line (e.g. `markbind serve --log=DEBUG` or something).

Hmm, true. This should definitely be the first step.

Apologies, what I meant more for this was, for example, printing the state of the `content` being passed down the promise chain in `page.generate()` after each significant stage, then outputting it into some files. This would allow for checking the output after each stage is as intended.
Not something suitable for printing in the console at all in hindsight.

I've edited the main post to reflect it

**Comment by yamgent** [comment]
Ok I get what you mean now, though since this is a developer feature for an existing command, I don't think we need a separate command exclusively for such a feature, a switch for developer mode for `serve` is sufficient (having this extra command doesn't help in not polluting the logic for site generation anyway).

> * performance logging of various stages of site generation

A built-in profiler would definitely be useful in gauging performance pain points, so I definitely see a value in it.

> * Outputting the state of `content` as it is passed through the various stages in `page.generate()` to seperate files

A debugger already allow you to examine the content though, so I am not sure whether this is inherently useful.

> * developer only front end components ( e.g. including a theme switcher on the template site to test front end changes for different themes )

This seems like a lot of work for a small bit of convenience?

**Comment by ang-zeyu** [comment]
> > * Outputting the state of `content` as it is passed through the various stages in `page.generate()` to seperate files
> 
> A debugger already allow you to examine the content though, so I am not sure whether this is inherently useful.
> 
> > * developer only front end components ( e.g. including a theme switcher on the template site to test front end changes for different themes )
> 
> This seems like a lot of work for a small bit of convenience?

Good point, for this particular use case, perhaps this could be a seperate issue "support live reloading for theme option in site.json"

> Ok I get what you mean now, though since this is a developer feature for an existing command, I don't think we need a separate command exclusively for such a feature, a switch for developer mode for serve is sufficient (having this extra command doesn't help in not polluting the logic for site generation anyway). 

> A built-in profiler would definitely be useful in gauging performance pain points, so I definitely see a value in it.

👍 
To sum, I'll rename this issue as "support performance logging option for markbind serve", if its fine!
